modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators/GlobalClassAnalyzer.cs

```diff
@@ -0,0 +1,103 @@
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Godot.SourceGenerators
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class GlobalClassAnalyzer : DiagnosticAnalyzer
+    {
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
+            => ImmutableArray.Create(
+                Common.GlobalClassMustImplementGodotObjectRule,
+                Common.GlobalClassMustNotBeGenericRule,
+                Common.ParentClassMustBeGlobalRule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.EnableConcurrentExecution();
+            context.RegisterSyntaxNodeAction(AnalyzeNode, SyntaxKind.ClassDeclaration);
+        }
+
+        private void AnalyzeNode(SyntaxNodeAnalysisContext context)
+        {
+            // Ignore syntax inside comments
+            if (IsInsideDocumentation(context.Node))
+                return;
+
+            var typeClassDecl = (ClassDeclarationSyntax)context.Node;
+
+            if (context.ContainingSymbol is not INamedTypeSymbol typeSymbol)
+                return;
+
+            // Search attributes for [GlobalClass] attribute
+            var attrs = typeSymbol?.GetAttributes();
+            if (attrs == null)
+                return;
+
+            bool isGlobal = false;
+            foreach (var attribute in attrs)
+            {
+                if (!(attribute.AttributeClass?.IsGodotGlobalClassAttribute() ?? false))
+                    continue;
+
+                isGlobal = true;
+                break;
+            }
+
+            if (!isGlobal)
+                return;
+
+            if (typeSymbol?.IsGenericType ?? false)
+                Common.ReportGlobalClassMustNotBeGeneric(context, typeClassDecl, typeSymbol);
+
+            if (!typeSymbol.InheritsFrom("GodotSharp", GodotClasses.GodotObject))
+            {
+                Common.ReportGlobalClassMustImplementGodotObject(context, typeClassDecl, typeSymbol!);
+                // Return since there is a chance that there is no parent.
+                return;
+            }
+
+            // I'd prefer if the GodotSharp stuff was already pre-annotated with [GlobalClass]...
+            // Suppresses a stupid 'editor\create_dialog.cpp:266 - Condition "inherits.is_empty()" is true.'
```

Comment:
The types in the GodotSharp assembly are proxy types for the native Godot classes, they aren't scripts. The global class feature is a script feature.

This might be a bug in the `create_dialog` or in `csharp_script`, I'm not aware of this being reported already so feel free to open a new issue.